### PR TITLE
[NTOSKRNL] Fix returned number of handles for Idle System Process

### PR DESCRIPTION
**Pull request by KRosUser**
## Purpose

As explained by @HBelusca in the Jira ticket, PsIdleProcess and PsInitialSystemProcess share the same HandleTable. This leads to the same number of Handles to be reported by ObGetProcessHandleCount then feeding the SYSTEM_PROCESS_INFORMATION structure?

JIRA issue: [CORE-16577](https://jira.reactos.org/browse/CORE-16577)

## Proposed changes

Proposed change is to force the HandleCount to 0 for the Idle process in SystemProcessInformation handling instead of calling ObGetProcessHandleCount.

## After change :

image

### CODE REVIEW

File: ntoskrnl/ob/obhandle.c [hunk 1]
```diff
@@ -60,6 +60,11 @@ ObGetProcessHandleCount(IN PEPROCESS Process)
 
     ASSERT(Process);
 
+    /* Idle Process shares the HandleTable with PsInitialSystemProcess so let's return these
```
Comment (HBelusca) [inline ntoskrnl/ob/obhandle.c:63]: Please move this to the ObGetProcessHandleCount call in the SystemProcessInformation handling. This is a hack needed only there.

Comment (KRosUser) [reply ntoskrnl/ob/obhandle.c]: Done

File: ntoskrnl/ex/sysinfo.c [hunk 2]
```diff
@@ -1031,7 +1031,9 @@ QSI_DEF(SystemProcessInformation)
                 SpiCurrent->BasePriority = Process->Pcb.BasePriority;
                 SpiCurrent->UniqueProcessId = Process->UniqueProcessId;
                 SpiCurrent->InheritedFromUniqueProcessId = Process->InheritedFromUniqueProcessId;
-                SpiCurrent->HandleCount = ObGetProcessHandleCount(Process);
+                /* Idle Process shares the HandleTable with PsInitialSystemProcess so let's return these
+                Handle count for System only, not Idle.	*/
+                SpiCurrent->HandleCount = (Process == PsIdleProcess)?0:ObGetProcessHandleCount(Process);
```
Comment (HBelusca) [inline ntoskrnl/ex/sysinfo.c:1036]:
```suggestion

                /* PsIdleProcess shares its handle table with PsInitialSystemProcess,
                 * so return the handle count for System only, not Idle one. */
                SpiCurrent->HandleCount = (Process == PsIdleProcess) ? 0 : ObGetProcessHandleCount(Process);

```

**Review by HBelusca — APPROVED** [review]

**Review by ThFabba — APPROVED** [review]

**Comment by KRosUser** [comment]
Thank you @ThFabba and @HBelusca 
@HBelusca : Shall we merge ?

**Pull request merged**